internal/providers/configdrive/configdrive.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configdrive
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+
+	"github.com/coreos/coreos-metadata/internal/providers"
+)
+
+var jsonmap map[string]interface{}
```

Comment:
Global variables should be avoided at almost all cost. This one is pretty easy to keep local.